src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestMessage.cs

```diff
@@ -40,6 +40,11 @@ public Version Version
                 ArgumentNullException.ThrowIfNull(value);
                 CheckDisposed();
 
+                if (value != HttpVersion.Version10 && value != HttpVersion.Version11 && value != HttpVersion.Version20 && value != HttpVersion.Version30)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), SR.net_http_request_invalid_version);
+                }
+
```

Comment:
Is it extensible for future? What if a third-party http client implementation that accepts higher versions?

Reply:
> Is it extensible for future?

What do you mean by extensible ? Is it handling new versions as they come off ? or giving users the choice to choose the versions they want ?

> What if a third-party http client implementation that accepts higher versions?

See https://github.com/dotnet/runtime/issues/72185#issuecomment-1396956555
> Question is whether that could break someone. For instance, someone asking for 2.5 and getting HTTP/2.

> If we'd chose to go the strict way, we should probably get it in early in the release to see if someone gets borked by it.

This change is made as an experiment, if people complain about it it will probably be rolled back

Reply:
> Is it handling new versions as they come off ? or giving users an api so they choose the versions they want ?

`HttpRequestMessage` is decoupled from `HttpClient` and can be used by others as well. I can assume when a new HTTP version comes out in the future there will be some 3rd implementations using the `HttpRequestMessage`, but this change will block them reusing the built-in type.

Reply:
Do you prefer doing higher validation in HttpClient.Send methods or below the stack in Socket/Connection Handlers ?

Reply:
I agree, this should be handler at the specific handler level. You can check out https://github.com/dotnet/runtime/blob/0588f245dba6418349c22894b40337afc34f1cfc/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectionPool/HttpConnectionPool.cs#L399 for Sockets handler.